eng/Version.Details.xml

```diff
@@ -41,6 +41,10 @@
       <Uri>https://github.com/dotnet/efcore</Uri>
       <Sha>b0b202671f7879070423e95776edc014885335ad</Sha>
     </Dependency>
+    <Dependency Name="Microsoft.Extensions.Telemetry.Testing" Version="8.0.0-preview.6.23320.3">
```

Comment:
If this is from the `main` branch, then everything's set up correctly in this PR. All we'll need is to add a subscription from extensions into this repo. The only issue there is it creates a circular dependency between this repo and extensions, which can be a problem for product construction. I think it'll be fine if we set up the `extensions -> aspnetcore` sub to be `everyWeek` frequency - @mmitche to confirm

Reply:
You beat me to it :P, I was about to ask about the circular dependency for this. Keep in mind this dependency is for use in tests only, so I think that in the case that darc doesn't allow to setup this subscription could go with two options:
- Either add support in dependency flow for test-only dependencies (similar to how we have special case for tooling today) so we can introduce ciruclar dependencies on these cases, or
- Not use dependency flow for test-only dependencies that could introduce ciruclar dependencies for repos. If we go with that plan, then we undo the change to this Version.Details.xml file and we just update the version we depend on in a as-needed basis.

Reply:
Or C: Do not allow circular dependencies.

Reply:
Circular dependencies are allowed for testing purposes. However, this must be moved to the ToolsetDependencies section otherwise the product is incoherent and cannot be shipped. What this means is:

- dotnet/extensions outputs will not show in the build drop.
- dotnet/extensions outputs is not part of the main SDK-umbrella product.

It's my understanding that this is the intention. If it's not, then we have a product layering issue, or we need to change how we prep the product for shipping. As usual, that would make me question why we have a separate repo for these things in the first place.

Reply:
Yup, the intention is for this to be test-only and shouldn't be in the build drop nor part of the main SDK product. @mmitche  so is the concrete feedback then that we should move these 4 lines on this file down to `<ToolsetDependencies>` section (line 360), and then that should allow us to setup darc subscription correctly? Or is there something else that needs to be done to setup the darc subscription?

Reply:
Yep, move the dependency. Subscriptions should be set so that they do not cycle endlessly. i.e. _at least_ one subscription in the cycle should be set to a periodic basis (e.g. weekly) rather than every build.

Otherwise all good.

Reply:
@JamesNK @joperezr let me know if you need my help setting up the subscription

Reply:
Yes please.

Note: The InstrumentRecorder has just been removed from runtime (for this preview) so this PR needs to make it in, otherwise we'll be blocked when consuming runtime updates.

https://github.com/dotnet/runtime/pull/87873

Reply:
To confirm, this dependency needs to be moved before this gets checked in.

Reply:
Subscription created 

```
https://github.com/dotnet/extensions (.NET 8) ==> 'https://github.com/dotnet/aspnetcore' ('main')
  - Id: ba55bbf2-4239-468a-a9c6-08db6772cb2c
  - Update Frequency: EveryWeek
  - Enabled: True
  - Batchable: False
  - PR Failure Notification tags: 
  - Merge Policies:
    AllChecksSuccessful
      ignoreChecks = 
                       [
                         "aspnetcore-quarantined-pr",
                         "aspnetcore-quarantined-pr (Tests: Helix),",
                         "aspnetcore-quarantined-pr (Tests: macOS),",
                         "aspnetcore-quarantined-pr (Tests: Ubuntu x64),",
                         "aspnetcore-quarantined-pr (Tests: Windows x64)"
                       ]
    NoRequestedChanges
```

Reply:
> To confirm, this dependency needs to be moved to Toolset before this gets checked in.

Moved dependency to toolset.